Shared/Supporting Files/Views/SampleLink.swift

```diff
@@ -0,0 +1,135 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct SampleLink: View {
+    /// The sample to present.
+    private let sample: Sample
+    
+    /// The text to bold in the sample's name and description.
+    private let textToBold: String
+    
+    /// Creates a link that presents a given sample.
+    /// - Parameters:
+    ///   - sample: The sample to present.
+    ///   - textToBold: The text to bold in the sample's name and description.
+    init(_ sample: Sample, textToBold: String = "") {
+        self.sample = sample
+        self.textToBold = textToBold
+    }
+    
+    var body: some View {
+        NavigationLink {
+            SampleDetailView(sample: sample)
+                .id(sample.name)
+        } label: {
+            SampleRow(
+                name: sample.name.boldingFirstOccurrence(of: textToBold),
+                description: sample.description.boldingFirstOccurrence(of: textToBold)
+            )
+        }
+    }
+}
+
+private extension SampleLink {
+    struct SampleRow: View {
+        /// The name of the sample.
+        private let name: String
+        
+        /// The name of the sample with attributes.
+        private let attributedName: AttributedString
+        
+        /// The description of the sample with attributes.
+        private let attributedDescription: AttributedString
+        
+        /// A Boolean value indicating whether the sample's description is showing.
+        @State private var isShowingDescription = false
+        
+        /// The names of the favorite samples loaded from user defaults.
+        @AppStorage(.favoriteSamplesKey) private var favoriteNames: [String] = []
+        
+        /// A Boolean value indicating whether the sample is a favorite.
+        private var sampleIsFavorite: Bool {
+            favoriteNames.contains(name)
+        }
+        
+        init(name: AttributedString, description: AttributedString) {
+            self.name = String(name.characters)
+            self.attributedName = name
+            self.attributedDescription = description
+        }
+        
+        var body: some View {
+            HStack {
+                VStack(alignment: .leading) {
+                    Text(attributedName)
+                    
+                    if isShowingDescription {
+                        Text(attributedDescription)
+                            .font(.caption)
+                            .opacity(0.75)
+                            .transition(.move(edge: .top).combined(with: .opacity))
+                    }
+                }
+                Spacer()
+                
+                if sampleIsFavorite {
+                    Image(systemName: "star.fill")
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(width: 12)
```

Comment:
This size looks better
```suggestion
                        .frame(width: 18)
```
![Simulator Screenshot - iPhone 15 Pro Max - 2023-12-13 at 16 29 31](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/4164028/ea890a4f-2a5e-4b8d-8b8c-4b73072eb629)